File: source/includes/avs-examples/index-management/update-index/edit-index.go
package main

import (
	"context"
	"fmt"
	"log"

	"go.mongodb.org/mongo-driver/mongo"
	"go.mongodb.org/mongo-driver/mongo/options"
)

func main() {

	ctx := context.TODO()

	// Replace the placeholder with your Atlas connection string
	const uri = "<connection-string>"

	// Connect to your Atlas cluster
	clientOptions := options.Client().ApplyURI(uri)
	client, err := mongo.Connect(ctx, clientOptions)
	if err != nil {
		log.Fatal(err)
	}
	defer client.Disconnect(ctx)

	// Set the namespace
	coll := client.Database("<databaseName>").Collection("<collectionName>")
	indexName := "<indexName>"

	type vectorDefinitionField struct {
		Type          string `bson:"type"`
		Path          string `bson:"path"`
		NumDimensions int    `bson:"numDimensions"`
		Similarity    string `bson:"similarity"`
	}

	type vectorDefinition struct {
		Fields []vectorDefinitionField `bson:"fields"`
	}

	definition := vectorDefinition{
		Fields: []vectorDefinitionField{{
			Type:          "vector",
			Path:          "<fieldToIndex>",
			NumDimensions: <numberOfDimensions>,
			Similarity:    "euclidean | cosine | dotProduct"}},
	}
	coll.SearchIndexes().UpdateOne(ctx, indexName, definition)

	if err != nil {
		panic(err)
	}

	fmt.Println("Successfully updated the search index")
}

File: source/includes/avs-examples/index-management/update-index/edit-index.cs
namespace query_quick_start;

using MongoDB.Bson;
using MongoDB.Driver;

public class IndexService
{
    private const string MongoConnectionString = "<connectionString>";
    // Other class methods here...
    public void EditVectorIndex()
    {
        try
        {
            // Connect to your Atlas deployment
            var client = new MongoClient(MongoConnectionString);

            // Access your database and collection
            var database = client.GetDatabase("<databaseName>");
            var collection = database.GetCollection<BsonDocument>("<collectionName>");
            
            var definition = new BsonDocument
            {
                { "fields", new BsonArray
                    {
                        new BsonDocument
                        {
                            { "type", "vector" },
                            { "path", "<fieldToIndex>" },
                            { "numDimensions", <numberOfDimensions> },
                            { "similarity", "euclidean | cosine | dotProduct" }
                        }
                    }
                }
            };
            
            // Update your search index
            var searchIndexView = collection.SearchIndexes;
            searchIndexView.Update(name, definition);
        }
        catch (Exception e)
        {
            Console.WriteLine($"Exception: {e.Message}");
        }
    }
}
